### ISSUE TITLE
Push constant in render bundle is panicking 

### ISSUE THREAD

**Issue by andrew-pa** [gfx-rs/wgpu#1062]
Here's the relevant stack trace:

```
thread 'main' panicked at 'range end index 16 out of range for slice of length 0', C:\Users\andre\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-core-0.6.5\src\command\bundle.rs:236:43
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/212e76c31d461462544b5d516d3bdc99fc8f016e\/library\std\src\panicking.rs:483
   1: core::panicking::panic_fmt
             at /rustc/212e76c31d461462544b5d516d3bdc99fc8f016e\/library\core\src\panicking.rs:85
   2: core::slice::index::slice_end_index_len_fail
             at /rustc/212e76c31d461462544b5d516d3bdc99fc8f016e\/library\core\src\slice\index.rs:41
   3: core::slice::index::{{impl}}::index<u32>
             at C:\Users\andre\.rustup\toolchains\beta-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\slice\index.rs:311
   4: core::slice::index::{{impl}}::index<u32,core::ops::range::Range<usize>>
             at C:\Users\andre\.rustup\toolchains\beta-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\slice\index.rs:15
   5: alloc::vec::{{impl}}::index<u32,core::ops::range::Range<usize>>
             at C:\Users\andre\.rustup\toolchains\beta-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\vec.rs:1982
   6: wgpu_core::command::bundle::RenderBundle::execute<gfx_backend_vulkan::Backend>
             at C:\Users\andre\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-core-0.6.5\src\command\bundle.rs:236
   7: wgpu_core::hub::Global<wgpu_core::hub::IdentityManagerFactory>::command_encoder_run_render_pass_impl<wgpu_core::hub::IdentityManagerFactory,gfx_backend_vulkan::Backend>
             at C:\Users\andre\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-core-0.6.5\src\command\render.rs:1501
   8: wgpu_core::hub::Global<wgpu_core::hub::IdentityManagerFactory>::command_encoder_run_render_pass<wgpu_core::hub::IdentityManagerFactory,gfx_backend_vulkan::Backend>
             at C:\Users\andre\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-core-0.6.5\src\command\render.rs:401
   9: wgpu::backend::direct::{{impl}}::command_encoder_end_render_pass
             at C:\Users\andre\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.6.0\src\backend\direct.rs:1237
  10: wgpu::{{impl}}::drop
             at C:\Users\andre\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.6.0\src\lib.rs:2257
  11: core::ptr::drop_in_place<wgpu::RenderPass>
             at C:\Users\andre\.rustup\toolchains\beta-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ptr\mod.rs:175

...
```

I've stepped through the code in a debugger and it looks like at some point the render pass forgets about the push constant data stored in it (`pass.base.push_constant_data`), it does set it in `RenderBundleEncoder::set_push_constants` but then by the time the `RenderBundle` gets executed, the equivalent vector of push constant data (`self.base.push_constant_data` at the location of the panic) is empty again, and so it panics. This is my first time with wpgu (which is otherwise awesome) and so I'm not sure if I'm doing something wrong or if it's broken, but so far as I can see it looks correct in my code. Perhaps this also might be a bug deeper in wgpu, I'm not sure.

**Comment by cwfitzgerald** [comment]
Do you have an easy repro case for this? A possibility would be [recording a trace](https://github.com/gfx-rs/wgpu/wiki/Debugging-wgpu-Applications#tracing-infrastructure).

**Comment by andrew-pa** [comment]
Yeah I could probably do that sometime. I did actually come up with a patch that seemed like it worked, but I wasn't quite sure how/where to contribute it back, or if it really matched the intent of the code or broke anything else. The commit is [here](https://github.com/andrew-pa/wgpu/commit/b28c987aaeccbe56b381b7a8500dbd145968afed) I just made it on the v0.6 branch so that I could get cargo to recognize it as a patched version. I haven't been back to work on this for a bit so it's not super high priority or anything. I'd be interested in actually contributing though if it's not overly complex.
